### PR TITLE
Add support for Chaz

### PR DESCRIPTION
**Pull request by jasonhazel**
## Description

Adding support for Chaz.

Had a previous PR for this board (#21091) but after seeing #21186 I closed it rework for RP2040 configurator support.  

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/hazel/chaz/keymaps/jasonhazel/keymap.json [hunk 1]
```diff
@@ -0,0 +1,128 @@
+{
+  "version": 1,
+  "notes": "",
+  "documentation": "",
+  "keyboard": "hazel/chaz",
+  "keymap": "default",
```
Comment (dunk2k) [inline keyboards/hazel/chaz/keymaps/jasonhazel/keymap.json:6]: would you be able to change `"default"` to `"jasonhazel"` so this file matches directory it's located under? 🙂

File: keyboards/hazel/chaz/promicro/info.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{
+    "manufacturer": "Hazel",
+    "keyboard_name": "Chaz",
+    "maintainer": "jasonhazel",
```
Comment (dunk2k) [inline keyboards/hazel/chaz/promicro/info.json:4]: `manufacturer`, `keyboard_name`, and `maintainer` lines are already specified in `info.json` in directory above, so they can be removed from this file.

File: keyboards/hazel/chaz/rp2040ce/info.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
+{
+    "manufacturer": "Hazel",
+    "keyboard_name": "Chaz",
+    "maintainer": "jasonhazel",
+    "url": "https://github.com/hazels-garage/chaz",
+    "diode_direction": "COL2ROW",
```
Comment (dunk2k) [inline keyboards/hazel/chaz/rp2040ce/info.json:6]: `manufacturer`, `keyboard_name`, `maintainer`, and `url` lines are already specified in `info.json` in directory above, so they can be removed from this file.

File: keyboards/hazel/chaz/rp2040ce/info.json [hunk 4]
```diff
@@ -0,0 +1,65 @@
+{
+    "manufacturer": "Hazel",
+    "keyboard_name": "Chaz",
+    "maintainer": "jasonhazel",
+    "url": "https://github.com/hazels-garage/chaz",
```
Comment (dunk2k) [inline keyboards/hazel/chaz/rp2040ce/info.json:5]: `manufacturer`, `keyboard_name`, `maintainer`, and `url` lines are already specified in info.json in directory above, so they can be removed from this file.

File: keyboards/hazel/chaz/rp2040ce/info.json [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    "matrix_pins": {
+        "cols": ["GP9", "GP8", "GP7", "GP6", "GP5"],
+        "rows": ["GP0", "GP2", "GP4", "GP28", "GP1", "GP3", "GP27", "GP29"]
+    },
+    "processor": "RP2040",
+    "bootloader": "rp2040",
+    "usb": {
+        "vid": "0x4A48",
+        "pid": "0x435A",
```
Comment (dunk2k) [inline keyboards/hazel/chaz/rp2040ce/info.json:19]: `vid` and `pid` are identical for both `rp2040ce` and `promicro` directories, would you be able to move these two lines to `info.json` in the directory above (keyboard level)?

File: keyboards/hazel/chaz/promicro/info.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        "nkro": false
+    },
+    "matrix_pins": {
+        "cols": ["B5", "B4", "E6", "D7", "C6"],
+        "rows": ["D3", "D1", "D4", "F5", "D2", "D0", "F6", "F4"]
+    },
+    "development_board": "promicro",
+    "usb": {
+        "vid": "0x4A48",
+        "pid": "0x435A",
```
Comment (dunk2k) [inline keyboards/hazel/chaz/promicro/info.json:18]: `vid` and `pid` are identical for both `rp2040ce` and `promicro` directories, would you be able to move these two lines to `info.json` in the directory above (keyboard level)?

Comment (dunk2k) [inline keyboards/hazel/chaz/promicro/info.json]: `vid` and `pid` are identical for both rp2040ce and promicro directories, would you be able to move these two lines to `info.json` in the directory above (keyboard level)?

File: keyboards/hazel/chaz/rp2040ce/info.json (hunk 5 above)
Comment (dunk2k) [inline keyboards/hazel/chaz/rp2040ce/info.json]: `vid` and `pid` are identical for both `rp2040ce` and `promicro` directories, would you be able to move these two lines to `info.json` in the directory above (keyboard level)?

File: keyboards/hazel/chaz/rp2040ce/info.json (hunk 4 above)
Comment (dunk2k) [inline keyboards/hazel/chaz/rp2040ce/info.json]: `manufacturer`, `keyboard_name`, `maintainer`, and `url` lines are already specified in `info.json` in directory above, so they can be removed from this file.

File: keyboards/hazel/chaz/promicro/info.json [hunk 7]
```diff
@@ -0,0 +1,64 @@
+{
+    "manufacturer": "Hazel",
+    "keyboard_name": "Chaz",
+    "maintainer": "jasonhazel",
+    "url": "https://github.com/hazels-garage/chaz",
```
Comment (dunk2k) [inline keyboards/hazel/chaz/promicro/info.json]: `manufacturer`, `keyboard_name`, `maintainer`, and `url` lines are already specified in `info.json` in directory above, so they can be removed from this file.

File: keyboards/hazel/chaz/keymaps/jasonhazel/keymap.json (hunk 1 above)
Comment (dunk2k) [inline keyboards/hazel/chaz/keymaps/jasonhazel/keymap.json]: would you be able to change `"default"` to `"jasonhazel"`, on line 6, so this file matches directory it's located under? 🙂

**Comment by jasonhazel** [comment]
I saw some comments come through relating to info.json information duplicated across files.  I went to reply and they're missing now so I assume a single, all encompassing comment is coming?

There is a lot of overlap between the two version info.json.  Is it safe to assume all of that overlap can be moved to the top level info.json?

File: keyboards/hazel/chaz/rp2040ce/info.json [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+        "device_version": "1.0.0"
+    },
+    "layouts": {
```
Comment (dunk2k) [inline keyboards/hazel/chaz/rp2040ce/info.json]: the layouts specified in `promicro` and `rp2040ce` are identical. Could you move "layouts" to `info.json` in directory above (keyboard level)? if done previously, please remove.

File: keyboards/hazel/chaz/promicro/info.json [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        "cols": ["B5", "B4", "E6", "D7", "C6"],
+        "rows": ["D3", "D1", "D4", "F5", "D2", "D0", "F6", "F4"]
+    },
+    "development_board": "promicro",
+    "usb": {
+        "vid": "0x4A48",
+        "pid": "0x435A",
+        "device_version": "1.0.0"
+    },
+    "layouts": {
```
Comment (dunk2k) [inline keyboards/hazel/chaz/promicro/info.json]: the layouts specified in `promicro` and `rp2040ce` are identical. Could you move "layouts" to `info.json` in directory above (keyboard level)?

**Comment by dunk2k** [comment]
@jasonhazel 

> There is a lot of overlap between the two version info.json. Is it safe to assume all of that overlap can be moved to the top level info.json?

you got it chief! There's argument for moving/merging the readme.md's to keyboard level, but I'll let the collaborators call that one 😉

**Comment by jasonhazel** [comment]
@dunk2k right on.  I applied your suggestions and will wait to see what the collaborators say about the readme. Thanks for your feedback!

**Review by dunk2k — APPROVED** [review]

File: keyboards/hazel/chaz/keymaps/default/keymap.json [hunk 10]
```diff
@@ -0,0 +1,128 @@
+{
+  "version": 1,
+  "notes": "",
+  "documentation": "",
```
Comment (waffle87) [inline keyboards/hazel/chaz/keymaps/default/keymap.json:4]:
```suggestion
```

File: keyboards/hazel/chaz/keymaps/jasonhazel/config.h [hunk 11]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2023 Jason Hazel (@jasonhazel)
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+#define TAPPING_TERM 200
```
Comment (waffle87) [inline keyboards/hazel/chaz/keymaps/jasonhazel/config.h:5]:
```suggestion
```

File: keyboards/hazel/chaz/keymaps/jasonhazel/keymap.json [hunk 12]
```diff
@@ -0,0 +1,128 @@
+{
+  "version": 1,
+  "notes": "",
+  "documentation": "",
```
Comment (waffle87) [inline keyboards/hazel/chaz/keymaps/jasonhazel/keymap.json:4]:
```suggestion
```

File: keyboards/hazel/chaz/promicro/readme.md [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+* Hardware Supported: Chaz, atmega32u4 ProMicro Compatible Controller
+* Hardware Availability: [Hazel's Garage](https://shop.hazel.cc/products/chaz), [Open Source](https://github.com/hazels-garage/chaz)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make hazel/chaz:default
+
+Flashing example for this keyboard:
+
+    make hazel/chaz:default:flash
```
Comment (waffle87) [inline keyboards/hazel/chaz/promicro/readme.md:17]: This file could just be moved to the top level keyboard directory, and the `keyboards/hazel/chaz/rp2040ce/readme.md` one removed.
```suggestion
* Hardware Supported: Chaz, atmega32u4 ProMicro Compatible Controller, RP2040 Community Edition Compatible Controller
* Hardware Availability: [Hazel's Garage](https://shop.hazel.cc/products/chaz), [Open Source](https://github.com/hazels-garage/chaz)

Make example for this keyboard (after setting up your build environment):

    make hazel/chaz:default
    make hazel/chaz/rp2040ce:default

Flashing example for this keyboard:

    make hazel/chaz:default:flash
    make hazel/chaz/rp2040ce:default:flash
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

File: keyboards/hazel/chaz/rp2040ce/info.json [hunk 14]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (drashna) [inline keyboards/hazel/chaz/rp2040ce/info.json:1]: It's using the same pinout as the pro micro, correct?

**Comment by jasonhazel** [comment]
@drashna I realized converters could be specified via rules.mk.  I've rearranged things to a single top-level `info.json` with promicro pins specified and then a sub directory for `rp2040ce` with the converter defined in the `rules.mk`.

File: keyboards/hazel/chaz/rp2040ce/rules.mk [hunk 15]
```diff
@@ -0,0 +1 @@
+CONVERT_TO=rp2040_ce
```
Comment (fauxpark) [inline keyboards/hazel/chaz/rp2040ce/rules.mk:1]: Converters at keyboard level won't work with Configurator.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**